core/src/executor/fetch.rs

```diff
@@ -60,3 +62,73 @@ pub async fn fetch<'a>(
 
     Ok(rows)
 }
+
+#[derive(futures_enum::Stream)]
+pub enum Rows<I1, I2> {
+    Derived(I1),
+    Table(I2),
+}
+pub async fn fetch_relation<'a>(
+    storage: &'a dyn GStore,
+    table_factor: &'a TableFactor,
+    filter_context: &Option<Rc<FilterContext<'a>>>,
+) -> Result<impl TryStream<Ok = Row, Error = Error, Item = Result<Row>> + 'a> {
+    match table_factor {
+        TableFactor::Derived { subquery, .. } => {
+            let filter_context = filter_context.as_ref().map(Rc::clone);
+            let rows = select(storage, subquery, filter_context).await?;
+
+            Ok(Rows::Derived(rows))
+        }
+        TableFactor::Table { name, .. } => {
+            let rows = storage
+                .scan_data(crate::data::get_name(name)?)
+                .await?
+                .map_ok(|(_, row)| row);
+            let rows = stream::iter(rows);
+
+            Ok(Rows::Table(rows))
+        }
+    }
+}
+
```

Comment:
I am still doing test on it.

Maybe this kind of code may work.
We can consider to add `fetch_relation_columns` to `executor/fetch.rs`
```rust
pub async fn fetch_relation_columns(
    storage: &dyn GStore,
    table_factor: &TableFactor,
) -> Result<Vec<String>> {
    match table_factor {
        TableFactor::Table { name, .. } => {
            let table_name = get_name(name)?;

            fetch_columns(storage, table_name).await
        }
        /*
        TableFactor::Derived { query: SetExpr::Select , .. } => {
            panic!();
            // we can fill this
        }
        */
        _ => {
            panic!();
        }
    }
}

```